### PR TITLE
fix: fetch only alerts for unacknowledged events

### PR DESCRIPTION
**Pull request by blenzi**
Following https://github.com/pyronear/pyro-api/pull/268, this PR fixes the retrieval of alerts by fetching the alerts associated with unacknowledged events.  In addition:

- increase the number of maximum events from 5 to 10
- add `call_api` method and use to avoid code repetition
- protection against undefined variables (`raise ValueError`)

### CODE REVIEW

**Review by MateoLostanlen — DISMISSED** [review]
Love this PR, thanks a lot @blenzi

**Review by Akilditu — CHANGES_REQUESTED** [review]
Thanks a lot for the PR. 

In addition to the comments, please bear in mind that the new get_alerts_for_event() should be in production before merging this.

File: app/utils/_utils.py [hunk 1]
```diff
@@ -240,3 +244,34 @@ def build_historic_markers(dpt_code=None):
 
     # We gather all markers stored in the fire_markers list in a dl.LayerGroup object, which is returned
     return dl.LayerGroup(children=fire_markers, id="historic_fires_markers")
+
+
+def call_api(func: Callable, user_credentials: Dict[str, str]) -> Callable:
+    """Decorator to call API method and renew the token if needed. Usage:
+
+     result = call_api(my_func, user_credentials)(1, 2, verify=False)
```
[the remark this answers is not in the document]
Comment (blenzi) [reply app/utils/_utils.py]: Thanks, done

File: app/main.py [hunk 2]
```diff
@@ -250,8 +250,8 @@ def update_live_alerts_data(
     user_token = user_headers["Authorization"].split(" ")[1]
     api_client.token = user_token
 
-    # Fetch the last 5 unacknowledged events and associated alerts
-    live_events = pd.DataFrame(call_api(api_client.get_unacknowledged_events, user_credentials)())[-5:]
+    # Fetch unacknowledged events and associated alerts (up to 15 per events_
+    live_events = pd.DataFrame(call_api(api_client.get_unacknowledged_events, user_credentials)())
```
[the remark this answers is not in the document]
Comment (blenzi) [reply app/main.py]: Ok, limited to 10

**Review by Akilditu — APPROVED** [review]
Thanks a lot for the changes. 

Can you confirme then production API is up to date and then we merge.

**Pull request merged**